### PR TITLE
Create catalog permissions.

### PR DESCRIPTION
**Pull request by Joonpark13**
## Hey, I just made a Pull Request!

These permissions will be used to integrate catalog with the permissions framework.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-common/src/permissions.ts
Comment (mtlewis) [inline plugins/catalog-common/src/permissions.ts:58]: is "designate" the right word here?

**Review by mtlewis — APPROVED** [review]
Looks great!

File: plugins/catalog-common/api-report.md
Comment (freben) [inline plugins/catalog-common/api-report.md:36]: Could we make it so that this one is free of warnings and `(undocumented)`? I think that's a good goal going forward overall.

Totally fine to do that in a later PR though.

File: plugins/catalog-common/api-report.md
Comment (freben) [inline plugins/catalog-common/api-report.md:15]: This is actually "delete" rather than "unregister" right? At least that's how I picture it in my mind.

**Review by freben — APPROVED** [review]
Approving to unblock, but I think the unregister->delete thing is worth fixing before merge.

File: plugins/catalog-common/api-report.md
Comment (Joonpark13) [reply plugins/catalog-common/api-report.md:15]: Done.

**Pull request merged**
